### PR TITLE
connection in AMQPBackend is deprecated

### PR DESCRIPTION
**Pull request by core23**

### CODE REVIEW

**Comment by OskarStark** [comment]
Hmm, this is a protected var and could be used by someone else and in this case this will be a BC, but indeed, we should try to get rid of this variable then.

@Soullivaneuh what do you think and how do we should handle this?

**Comment by Soullivaneuh** [comment]
Indeed, although this is never used, it could be used on a child class.

If we remove it as is, this will produce some errors.

I think we should have a deprecated tag instead.

BTW, I'm curious of why this property is here. ping @rande

**Comment by rande** [comment]
a mistake from 2012 ;) https://github.com/sonata-project/SonataNotificationBundle/commit/e891e42ce7a44d92c93cffea5f0ee7e1ed46892f#diff-65be67a60425e32fb0392880b02aff19R34

**Comment by OskarStark** [comment]
Then i am for removing it with an upgrade notice and schedule this for 3.0 

What do you mean ?

**Comment by Soullivaneuh** [comment]
> Then i am for removing it with an upgrade notice and schedule this for 3.0

Yes, but with a deprecation for next minor. :-)

**Comment by core23** [comment]
Added deprecation info for now @Soullivaneuh

File: Backend/AMQPBackend.php [hunk 1]
```diff
@@ -30,6 +30,9 @@ class AMQPBackend implements BackendInterface
 
     protected $queue;
 
+    /**
+     * @deprecated This property is deprecated since version 2.3 and will be removed in 3.0.
```
Comment (Soullivaneuh) [inline Backend/AMQPBackend.php]: Since 2.4.

2.3 is already stable.

BTW, a new branch should be created before merging this PR.

**Comment by OskarStark** [comment]
@core23 can you please change the commit message? and create a new PR for the next major where you remove the property.

don't forget the upgrade notice

thank you

**Comment by OskarStark** [comment]
Restarted travis, there was a timeout.

When travis is green, this PR can me merged IMO. ping @Soullivaneuh

**Comment by core23** [comment]
RTM @OskarStark

**Comment by core23** [comment]
Thank you @rande

**Pull request merged**